libsolidity/ast/ASTJsonExporter.cpp

```diff
@@ -371,6 +371,7 @@ bool ASTJsonExporter::visit(EnumDefinition const& _node)
 	std::vector<pair<string, Json::Value>> attributes = {
 		make_pair("name", _node.name()),
 		make_pair("nameLocation", sourceLocationToString(_node.nameLocation())),
+		make_pair("documentation", _node.documentation() ? toJson(*_node.documentation()) : Json::nullValue),
```

Comment:
I feel like this should have a unit test, but I haven't been able to find one for both the exporter and the importer. How is this tested - is it done implicitly via `syntaxTests`?

Also, if there's no documentation in the node, i.e. if we set `"documentation"` to `Json::nullValue`, what will the output look like? Will there be a `"documentation"` key in the JSON, e.g. `"documentation": ""`, or will it be omitted?

Reply:
There are some trivial natspec tests where I've added at least some enum definition with natspec comments.

The way I am constructing this JSON node is exactly like it is done for other "documentation" nodes.

Reply:
We do have some natspec tests in [`SolidityNatspecJSON.cpp`](https://github.com/ethereum/solidity/blob/develop/test/libsolidity/SolidityNatspecJSON.cpp). These test the output of `--devdoc` and `--userdoc`, not the AST though. For the AST we could add an ASTJSON test. We also have the import/export covered by a script that exports and reimports the AST of all syntax and ASTJSON tests: https://github.com/ethereum/solidity/blob/8830be981725f522fa996a2d79c83e8a3ced76a7/scripts/ASTImportTest.sh#L100

I guess a test showing that an enum does not produce any devdoc or userdoc would be useful too though. It would show that this is the expected behavior and let us notice if it ever changes by mistake.